core/src/token_info/onchain.rs

```diff
@@ -0,0 +1,27 @@
+use anyhow::Result;
+use futures::future::{BoxFuture, FutureExt};
+
+use super::{TokenBaseInfo, TokenId, TokenInfoFetching};
+use crate::contracts::stablex_contract::StableXContractImpl;
+
+impl TokenInfoFetching for StableXContractImpl {
+    fn get_token_info<'a>(&'a self, id: TokenId) -> BoxFuture<'a, Result<TokenBaseInfo>> {
+        async move {
+            let info = self.get_token_info(id.into()).await?;
```

Comment:
ubernit: I would prefer destructuring here instead of indexing into the tuple:
```suggestion
            let (alias, decimals) = self.get_token_info(id.into()).await?;
```